docs/app/guides/accessibility-testing.mdx

```diff
@@ -0,0 +1,150 @@
+
+This approach explicitly avoids testing accessibility - which is all about the nature and structure of the content being tested, and how well the implementation in code matches that content, to provide a functional experience in a range of assistive technologies and browser setups.
+
+Testing Library (introduced in 2018 and now widely popular) is sometimes seen as a solution to the fact that explicit Test IDs do not provide any accessibility testing benefit. While Testing Library's helpers can be convenient and useful in their own right, and they keep accessibility top-of-mind for testers, it's important to note that Testing Library locators do **not** guarantee the accessibility of the elements being tested.
+
+:::warning
+Whether you primarily use Testing Library element locators, test IDs, or a different approach, testing for accessibility requires making specific assertions about the **implementation and behavior** of your application.
+:::
+
+There is much more to this topic than we can cover here, but we will provide a small example, and a recommendation for what to do.
+
+### Roles vs elements - button example
+
+The native HTML `button` element has a complex accessibility contract implemented by the browser for keyboard and screen reader users. Here is a partial list of ways a button element is automatically managed by the browser:
+
+- It has the implicit role of button so that users know what it is - a screen reader, for example, would announce it as a button
+- It has a default browser button style to visually distinguish it from surrounding content
+- It has default browser styles for hover and "pressed" states
+- It is placed in the tab order of the document and can be focused with the keyboard
+- When focused it receives a focus outline style
+- It can be activated with a mouse click, or the enter key, or the space bar
+- If placed inside a form element, activating the button by any of these methods will submit the form (because its default type is `submit`)
+- It will receive specific styling, customizable by the user, in Windows High Contrast Mode (used by people with certain visual disabilities)
+
+On the other hand, a `div` with `role` of `button` does not get this default browser accessibility behavior, and is not accessible without custom JavaScript and CSS replacing the browser contract.
+
+This means that, without some assertions about either the `button` element itself, or a full test of the accessibility "contract" of that element, it is not safe for a developer to refactor from from a `button` to a `<div role="button">`, even though Testing Library will treat each as the same if located using the `ByRole` locator. This is because, much like test IDs, the `ByRole` approach is intended to _avoid_ testing the implementation directly, to keep tests resilient.
```

Comment:
```suggestion
This means that, without some assertions about either the `button` element itself, or a full test of the accessibility "contract" of that element, it is not safe for a developer to refactor from a `button` to a `<div role="button">`, even though Testing Library will treat each as the same if located using the `ByRole` locator. This is because, much like test IDs, the `ByRole` approach is intended to _avoid_ testing the implementation directly, to keep tests resilient.
```